### ISSUE TITLE
[Build] Stride.Core.Tasks fails on pack-assets on multiple projects

### ISSUE THREAD

**Issue by hjred** [stride3d/stride#1051]
**Release Type**: GitHub

**Version**: master

**Platform(s)**: Windows

**Describe the bug**
When trying to build Stride.GameStudio the build fails with code 1 in Stride.Core.Tasks when pack-assets is executed for
* Stride.Graphics
* Stride.Rendering
* Stride.Engine
* Stride.Voxels
* Stride.Video
* Stride.Particles
* Stride.SpriteStudio.Offline
* Stride.Assets.Presentation
* Stride.Samples.Templates

**To Reproduce**
1. Installing prerequisites
2. git clone
3. Open Stride.sln
4. Build Stride.GameStudio

**Expected behavior**
Build executes sucessfully on clean run

**Additional context**
Visual Studio 2017 is **not** installed (and was never installed on that system).
Visual Studio 2019 16.10.0 Preview 2.0 is used.
Windows SDK 10.0.18362.0 and 10.0.19041.0 are installed.

**Comment by tebjan** [comment]
Can you confirm that gut lfs works properly? How did you check out the repo? Command line or GitExtensions?

**Comment by hjred** [comment]
git lfs is installed, i created a local test repository and initialized git lfs, which worked:

    git init
    git lfs install

I use the official git windows build (2.31.0) and cloned the Repo from git bash command line:

    git clone https://github.com/stride3d/stride.git

Is there another way to test if it had worked when cloning the repo?

**Comment by tebjan** [comment]
Not sure, tbh. I use GitExtensions, which automatically takes care of this. Check whether the dlls i the deps folder are there or just 1kb text files.

**Comment by hjred** [comment]
Yes, various dll's and executables are in the deps folder.

**Comment by tebjan** [comment]
But looking at the build error, you seem to miss the net5 SDK. Maybe the prerequisites are still for .net core 3.1.

**Comment by hjred** [comment]
dotnet --version

    5.0.300-preview.21180.15

dotnet --list-sdks

    3.1.408 [C:\Program Files\dotnet\sdk]
    5.0.300-preview.21180.15 [C:\Program Files\dotnet\sdk]

**Comment by manio143** [comment]
Can you try with a non-preview version of the .NET 5 SDK? We've had cases in the past where preview versions were causing new issues.

**Comment by xen2** [comment]
Stride needs .NET SDK 5.0.**100** installed.
We have a `global.json` to supposedly choose/enforce it, I thought it would complain at runtime if the version doesn't match.
Can you try to install it?

**Comment by xen2** [comment]
Sorry, I realize it's the Core.Tasks failing, which is loaded another way.
Let me check into that.

**Comment by xen2** [comment]
Can you try again with latest `master`?
(if possible first before installing the 5.0.100 SDK, to double-check if it displays a proper error message this time).

**Comment by hjred** [comment]
Yes will try with the current master first and then with non-preview build

**Comment by hjred** [comment]
After the update, opening Stride.sln in Visual Studio and trying to build Stride.GameStudio results in no noticable difference.
Building Core.Tasks finished without any errors and the same errors regarding the exit code are happening.

**Comment by hjred** [comment]
Installing a non-preview version of Visual Studio 2019 was not enough (which by the way includes Net SDK 5.0.202). The preview needed to be uninstalled otherwise there were still references to the preview SDK even though the build itself used SDK 5.0.202.
(tried to find some nuget assembly referenced in C:\Program Files\dotnet\sdk\5.0.300-preview.21180.15).

But now the build works, thanks a lot

**Comment by xen2** [comment]
It seems that global.json is not respected when running the `.exe`, only when using `dotnet` command.
We might have to change strategy (i.e. using `AssemblyLoadContext` to load either NuGet or MSBuild so that they can have separate versions)

**Comment by arturoc** [comment]
I'm getting this same issue with latest visual studio 2022 and latest master

**Comment by arturoc** [comment]
installing .net 6 SDK fixed it
